src/starknet_runner/runner.rs

```diff
@@ -32,6 +32,20 @@ impl<H> StarknetRunner<H>
 where
     H: SyscallHandler,
 {
+    pub fn map_hint_processor<H2>(
+        self,
+        hint_processor: SyscallHintProcessor<H2>,
+    ) -> StarknetRunner<H2>
+    where
+        H2: SyscallHandler,
+    {
+        StarknetRunner {
+            cairo_runner: self.cairo_runner,
+            vm: self.vm,
+            hint_processor,
+        }
+    }
+
```

Comment:
the purpose of the function is only to change the hint processor?

Reply:
Not only the `hint_processor`, but its type (the generic) too, which can't be done by just assigning to it.